pep-0654.rst

```diff
@@ -126,14 +126,24 @@ Specification
 ExceptionGroup
 --------------
 
-The new builtin exception type, ``ExceptionGroup`` is a subclass of
-``BaseException``, so it is assignable to ``Exception.__cause__`` and
-``Exception.__context__``, and can be raised and handled as any exception
-with ``raise ExceptionGroup(...)`` and ``try: ... except ExceptionGroup: ...``.
-
-Its constructor takes two positional-only parameters: a message string and a
-sequence of the nested exceptions, for example:
+We propose to add two new builtin exception types: ``ExceptionGroup`` and
+``BaseExceptionGroup``. The former is a subclass of ``Exception`` and
+the latter is a subclass of ``BaseException``. Both are assignable to
+``Exception.__cause__`` and ``Exception.__context__``, and they can be
+raised and handled as any exception with ``raise ExceptionGroup(...)`` and
+``try: ... except ExceptionGroup: ...`` or ``raise BaseExceptionGroup(...)``
+and ``try: ... except BaseExceptionGroup: ...``.
+
+Both have a constructor that takes two positional-only parameters: a message
+string and a sequence of the nested exceptions, for example:
 ``ExceptionGroup('issues', [ValueError('bad value'), TypeError('bad type')])``.
+The difference between them is that ``ExceptionGroup`` can only wrap
+``Exception`` subclasses while ``BaseExceptionGroup`` can wrap any
+``BaseException`` subclass. A factory method that inspects the nested
+execptions and selects between ``ExceptionGroup`` and ``BaseExceptionGroup``
+makes the choice automatic. In the rest of the document, unless stated
+otherwise, when we refer to ``ExceptionGroup`` we mean either an
+``ExceptionGroup`` or a ``BaseExceptionGroup``.
```

Comment:
This last thing is slightly awkward. Maybe we should write "exception group" when either can be used and "ExceptionGroup" when that specific type is meant? (Maybe do this in a separate PR if it affects a lot of locations in the doc, as I anticipate.)

Reply:
Agreed, let's fix and merge the two open PRs and then I'll change this in a new one.